include/arch/arm/cortex_m/exc.h

```diff
@@ -27,8 +27,7 @@ extern "C" {
 #define _ZERO_LATENCY_IRQS_RESERVED_PRIO 0
 #endif
 
-#if defined(CONFIG_CPU_CORTEX_M_HAS_PROGRAMMABLE_FAULT_PRIOS) || \
```

Comment:
Aren't those 2 completely different features? BASEPRI and programmable fault priorities? What if a future Arm variant comes with one but not the other?

Reply:
@carlescufi this is really hard to imagine. However, even in that weird scenario:
What the code does, inside this #ifdef guard, is to reserve one priority level for the FAULT priorities. Strictly speaking, this only requires that the FAULT exceptions exists and their priorities are programmable.
It has nothing to do with BASEPRI; BASEPRI is the _current_ way to mask-out a subset of priority levels.